File: platform/app/schemas/project.py
from __future__ import annotations

from datetime import datetime
from typing import List, Optional

from pydantic import BaseModel


class ProjectCreateRequest(BaseModel):
    name: str
    display_name: str
    repo_url: Optional[str] = None
    branch: str = "main"
    description: Optional[str] = None


class ProjectUpdateRequest(BaseModel):
    display_name: Optional[str] = None
    repo_url: Optional[str] = None
    branch: Optional[str] = None
    description: Optional[str] = None
    status: Optional[str] = None


class ProjectResponse(BaseModel):
    id: str
    name: str
    display_name: str
    repo_url: Optional[str] = None
    branch: str = "main"
    description: Optional[str] = None
    status: str = "active"
    tech_stack: Optional[List[str]] = None
    repo_tree: Optional[str] = None
    last_synced_at: Optional[datetime] = None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class ProjectListResponse(BaseModel):
    items: List[ProjectResponse]
    total: int


class ProjectSyncResponse(BaseModel):
    tech_stack: List[str]
    tree_depth: int
    readme_length: int
    synced_at: datetime

File: web/src/types/project.ts
export interface Project {
  id: string;
  name: string;
  display_name: string;
  repo_url: string | null;
  branch: string;
  description: string | null;
  status: 'active' | 'archived';
  tech_stack: string[] | null;
  repo_tree: string | null;
  last_synced_at: string | null;
  created_at: string;
  updated_at: string;
}

export interface ProjectListResponse {
  items: Project[];
  total: number;
}

export interface ProjectCreateRequest {
  name: string;
  display_name: string;
  repo_url?: string;
  branch?: string;
  description?: string;
}

export interface ProjectUpdateRequest {
  display_name?: string;
  repo_url?: string;
  branch?: string;
  description?: string;
  status?: string;
}

export interface ProjectSyncResponse {
  tech_stack: string[];
  tree_depth: number;
  readme_length: number;
  synced_at: string;
}
